### ISSUE TITLE
Sprint - June 24 to July 5, 2024

### ISSUE THREAD

**Issue by benjackwhite** [PostHog/posthog#23075]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by pauldambra** [comment]
# Team Error Tracking

**Support hero:** @daibhin   

* paul out a day

## items from previous sprint

### High priority

- hiring (@pauldambra) 🔴
- hogql filters performance (@pauldambra ) ✅
    - death by a 1000 cuts here... 
    - turned this on for $large-eu-customer now though
    - so we might be able to start deleting old code 🤞 now
- masking text in the screenshot images on Android and iOS @marandaneto
    - 
- Error tracking (@pauldambra and @daibhin)
    - first pass UI out
    - user interview booked with $large-eu-customer
    - some grouping research done but lots of context switching 😓 

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 🟡 quality and consolidation
    - mobile replay open beta
        - out free and being promoted in onboarding
    - replay captures every site
        - David working with rrweb to improve CSS parsing, already a major reduction in issues as a result
    - people can find the valuable recordings 
        - universal filters in progress
- 🟡 Error tracking MVP
    - started
- 🟡 Hire _at least_ 2 amazing colleagues
   
### High priority

- 

### Low priority / side quests

- mobile usage report stuff @pauldambra 
 - persistent replay queue in posthog-js @pauldambra 
 - icons left/right for android wireframes @daibhin 
 - we are recruiting beta testers for Android and iOS - (we should have screenshots now) @marandaneto @annikaschmid

**Comment by benjackwhite** [comment]
# Team DeliverHog

Ben + Marius OOO for second week of sprint (50% capacity)

## Retro

- BW - Felt super productive
- MA - Great progress all round

## Hang over items from previous sprint

- [x] Have the new "CDP" service in production, flagged to Team 2 @benjackwhite 
- [ ] Get all of our existing Zapier actions (e.g. slack notifications) running as hog functions (dog-fooding) @benjackwhite
  - [ ] Not done - we weren't really using actions for this, but multiple destinations setup
  - [ ] (Maybe by end of sprint we have Customer.io / hubspot working)
- [x] Missing hog features to support the above ([related issue](https://github.com/PostHog/meta/issues/208)) @mariusandra 
- [x] Make editing Hog not suck (syntax highlighting, autocomplete) 
- [x] no semicolons!
- [x] Brownbag on Hog and new CDP architecture

## OKR

1. 🔴 Get the new CDP in front of real users
- why red? Planned vacations mean we don't want to hand it out and walk away immediately - that said, we are really not far off and probably _could_ have hit this without vacation

## Sprint plan

Goal: Have DeliveryHog ready for Closed beta testing
(See https://github.com/PostHog/posthog/issues/22833)

- [ ] Scaling work @benjackwhite 
  - [ ] Overflow topic with detection for slow functions
  - [ ] Metrics, dashboards, alerts
- [ ] Secret management (probably go for simple encrypted field in django) @benjackwhite 
- [ ] Hook up to rusty webhook service (hopefully Brett, if not Marius)
- [ ] Ensure every existing plugin destination _could_ be built with Hog and build them as templates @mariusandra 
- [ ] Go through existing CodeQL reports and sanity check the safety of the system @mariusandra 

## Side quests
- [ ] Load testing of Hog functions in our own account
- [ ] Test execution with sample event
  - [ ] Estimate of throughput for a function (based on filters)
  - [ ] Example event to choose from events matching filters

**Comment by fuziontech** [comment]
# Team Click Haus

### OKR Q2 2024

### Objective

**James as a Service -> Clickhouse as a Service**

* **Key Results:**
    * Better Visibility
         * Regularly testing backups
         * Monitoring/alerting
         * Mutations
         * Moves
    * Management
         * Managing/killing mutations
    * Self Serve
         * Schema design feedback (James non blocking
         * Schema management   
    * Automation
         * Replace/Upgrade replicas
         * Upgrading to 24.04
         * Disk configs 
 
### Board
https://github.com/orgs/PostHog/projects/85/views/2

### Retro

**Infra**


- [x] 🔥  ZK incident followups
- [x] ✍️ Write up the outcome from CH strategy planning
- [x] 🏃 Onboarding Dani
- [x] ♻️ Replace offline nodes w/ nvme nodes
- [ ] 📟 Monitoring and Alerting on EU Coordinator
- [ ] 🧪 Test incremental backup restores
- [ ] ⏩ Move parts around so last 3 months of data are on NVME on US
- [ ] 🗑️ Delete persons on teams that are still ingesting data (for personless events)
~~- [ ] 👯 Debug duplicate events in `sharded_sessions` on EU (bad replication)~~

### High priority

**Infra**
- [ ] Retire old Offline Nodes on US Cluster @fuziontech 
- [ ] 🧪 Test incremental backup restores @Daesgar 
- [ ] ⏩ Move parts around so last 3 months of data are on NVME on US  @Daesgar 
- [ ] Remove projections in EU on events table @fuziontech 
- [ ] 🗑️ Delete persons on teams that are still ingesting data (for personless events) @fuziontech 
- [ ] Configs in Ansible for ClickHouse US and EU @Daesgar 
- [ ] 🏃 4 new im4gn.16xlarge replicas for US @fuziontech

**Comment by benjackwhite** [comment]
# Team Infra

## Retro
- Ben out for 50%
- Sašo out 1 day
- Frank out 100%

## Hang over items from previous sprint
- 🟢  SEIM: 
     - 🔴 access logs to django @benjackwhite (fix axes?)
     - 🟢 cloudtrail dashboard @danielxnj 
     - 🟢 eks audit dashboard @danielxnj 
     - 🟢 documentation/policy for sharing with auditors/clients (in cloud-infra?) @danielxnj 
       - https://github.com/PostHog/posthog-cloud-infra/issues/2808
     - 🟢 metabase logs into loki
- 🟡  secrets
    - 🟢 vault monitoring
    - 🟢 vault is all configured with TF
    - 🔴 backups @danielxnj
      - Decided to move to use a proper database backing which will take care of backups for us
    -  🔴 CDP integration @ZeleniJure 
      - opted against this as it isn't fit for purpose (scale we use it etc.)
- 🟡  VPA dashboard to understand over-provisioned resources @ZeleniJure 
  - Still last tweaks in prod to get it working (and then we need to implement the recommendations)
- 🟢 roll out new codified deployment strategy @frankh 
- 🟢 Other cost optimization stuff
  - Removed the overprovisioning deployments
  - Tagged Clickhouse disks (manual atm as it is still not in terraform)
  - S3 costs reducing thanks to incremental backups of clickhouse
- 🔴  internal RBAC review (tbc)
- 🟢 S3 mounted volume for Ingestion (geoip + blobby models)

## OKR

1. 💪 Deploy with confidence 🟢 
- Finalize our Canary Deploy process 🟡
5. 🚨 Improved alerting and monitoring 🟢 
- SIEM work has made this a priority
6. 🔒 Deeper Security 🟡 
9. 💰 Continued cost control 🟢 

### High priority
- [Q3 Planning](https://docs.google.com/document/d/1plqy2Yqf8wosKXvhoIDGk05NB86AlqS2RQ_qckS9tdY) @benjackwhite 
- Vault in use for production secrets (k8s configuration for example) @ZeleniJure @danielxnj 
  - why? We've done most of the thinking, now we just need to do it to complete our "auditable security quarter"
- Over-provisioned resource review @ZeleniJure 
  - why? Part of cost saving efforts to reduce our over provisioned deployments
- Audit all our AWS IAM accounts (we have everything in SSO so we shouldn't have any normal accounts there) @danielxnj

### Side quest
- access logs to django @danielxnj 
- Make sure all kafka topics are in IaC (e.g. clickhouse_heatmap_events)

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @dmarticus 
Days off:
Juraj:  5 days
Phani: 0 days
Dylan: 0 days
Neil: 2 days

## Retro


- release event based triggers - @Phanatic ✅
- Surveys branching logic - @jurajmajerik 🟡 (almost out, some secondary stuff remains)
- https://github.com/PostHog/posthog/issues/20851 - @dmarticus ❌
- https://github.com/PostHog/posthog/issues/22516 - @dmarticus ❌ - unexpectedly off more days, focused on getting delayed surveys shipped, an excellent refactor to make it all work well.
- Make a monorepo (https://github.com/PostHog/meta/pull/207/ ) - @neilkakkar ✅
- https://github.com/PostHog/posthog/issues/22131 continue on rewrite - @neilkakkar 

## Hang over items from previous sprint

---

## OKRs

1. Make sure feature flags can handle 10x current scale
2. Polish new experiments UI & collect feedback
3. Add most requested surveys functionality

### High priority

- event based triggers with actions and filters - @Phanatic 
- Surveys branching logic release - @jurajmajerik 
- no-code experiments RFC - @jurajmajerik 
- https://github.com/PostHog/posthog/issues/20851 - @dmarticus 
- https://github.com/PostHog/posthog/issues/22516 - @dmarticus 
- https://github.com/PostHog/posthog/issues/22131 continue on rewrite - db access - @neilkakkar 

### Low priority / side quests / maybe Neil will get to this next year

- Temporal queues for feature success - @neilkakkar 
- Setup instrumentation for flip-flopping problem of experiment significance - @neilkakkar

**Comment by EDsCODE** [comment]
# Team Data <->
### OKR Q2 2024

### Objective

**Release data warehouse to everyone**

* **Key Results:**
    * Integration first experience
         * schemas are reliable
         * modeling of each integration is clear
         * Good automatic roll up views and joins
         * Wizard to onboard people
    * Establish a solid pattern to build integrations
    * Complete data warehouse experience in the rest of the app (insights, feature flags, experiments)
 
### Retro

- [x] incremental syncs @Gilbert09 
- [ ] templated joins @EDsCODE 
- [x] UI revamp @EDsCODE 
- [ ] warehouse props in feature flags and experiments
- [x] person model batch exports @tomasfarias 


### High Priority

- [ ] postgres and other database incremental syncs @Gilbert09 
- [ ] onboarding flow @EDsCODE 
- [ ] errors from syncing and linking shown @EDsCODE 
- [ ] person model batch exports @tomasfarias 
- [ ] completing these todos for pricing release: https://github.com/PostHog/posthog/issues/22669

Side quests:
- [ ] warehouse props in feature flags and experiments
- [ ] templated joins

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

    - @raquelmsmith 
        - [x] Finish roll-out of personless events
            - Update migrate script to also report usage at time of migration
            - Roll out to 5-10k customers at a time 
            - Fix changelog link
            - Fix sentry errors
        - [ ] Referrals: add/edit referrer and redeemer
        - [ ] Start working on toolbar dashboard template thing
        - [x] Prune usage report queries
        - [x] Unexpected bill emails
    - @zlwaterfield 
    	- [ ] clean up outstanding feature flags (reverse-proxy-onboarding and email-verification-ticket-submission)
    	- [x] ship outstanding PRs (Teams addon - remove modal, Projects pay gate on free should be to paid not teams)
    	- [x] subscribe to all products
    		- main API changes (basic serializer set up, products=all query, default paid/free plan map, functions for upgrading/downgrading
    		- feature flag in UI on billing page with main button alterations (no comparison or other UI changes)
    	- [x] startup plan metadata RFC
    	- [ ] emails new teams addon customers
    	- [ ] re-run the plans map and compare with the new auto-cancel functionality 

</details>


## Q2 Goals
✅=finished 🟡=in progress 🔴=won't finish
1. 🟡 Create a flow in product analytics onboarding to fill out a dashboard template using actions (Raquel)
2. 🟢  Simplify our subscription flows (Zach, supported by Raquel)
    - ✅ Move teams plan to being an addon
    - 🟢 Change to a single subscribe for all products
4. ✅  Launch pricing changes (Raquel)
    - ✅   Personless events - will help us reach more customers at an affordable price

## This sprint

### High priority
- @raquelmsmith 
	- [ ] Support for first week 
    - [ ] Pricing page experiments - iterate here with cory and eli until it's done
    - [ ] Stay on top of revenue issues
    - [ ] Start working on toolbar dashboard template thing 
    - [ ] Keep on top of personless comms and customer issues and metrics
    - [ ] Lots of interviews...
- @zlwaterfield 
	- [ ] Complete _subscribe to all products_
		- [ ]  frontend changes
		- [ ] release under feature flag to new users
		- [ ] backfill existing users and communicate with them
		- [ ] (if time permits - probably next sprint) cleanup! remove/clean single product subscribe code where we can.
	- [ ] Start on the Stripe metadata changes - close RFC, updates to Zapier, work on backfill, etc.
	- [ ] re-run the plans map and compare with the new auto-cancel functionality

**Comment by bretthoerner** [comment]
# Team Pipeline

Off: Tiina
Support: Xavier

## Retro

### High priority

- [x] Improve ingestion efficiency by moving the last Kafka waits at the end of the batch (Brett)
- [x] New customers write to person only from $set and $identify, $create_alias, $merge_dangerously and new products also only read persons (Tiina)

## OKR

✅=finished 🟢=on track to finish this quarter 🟡=might not finish 🔴=won't finish
✔️=progressed last sprint ; ➡️=planned work for this sprint

🟢✔️ Deprecate posthog-events by moving to capture-rs fully (21659)
🟢✔️ Visibility into what's in the ingestion queues and past performance (20985)
🟢✔️ Fast configuration options to speed up incident recovery, e.g. by token send to overflow or drop (21662)
🟢✔️ Iterate on person processing to make it faster and cheaper (21048)
🟢✔️ Batch exports UX improvements, e.g. error notifications and UI rewamp (21139)
🟢 Support adding new products (21665)
🟢✔️ Deprecate scheduler & jobs deployments, runEveryX plugins and kafkajs consumers (21656)

### High priority

- [ ] Hog support for Rusty-Hook (Brett)
- [ ] capture-rs: fix billing limits (Xavier)

### Low priority / side quests

- [ ] Finish hog-rs to posthog repo migration (deploy out of posthog through state.yaml)
- [ ] Collect rdkafka metrics (broker response latency, error rates) for all node producers & consumer (Xavier)
- [ ] capture-rs: read redis out-of-band (avoid latency if redis slow)

**Comment by robbie-c** [comment]
# Team ~~web analytics~~ session table

**Support hero:** @robbie-c 

## Retro

Some personless events work (low priority tidying up of posthog-js) was blocked on https://github.com/PostHog/product-internal/pull/612/files, so I got to work on the sessions table instead. It's pretty close to being ready, maybe by the end of this sprint. PR: https://github.com/PostHog/posthog/pull/23023 . I've done the first 90%, just finishing off the second 90%

- 🟢 Start emailing customers for research (emailed 30, spoke to 5)
- 🟢 Posthog-js: Rewrite posthog-js person props backfilling to include initial current url
- 🛑 Posthog-js: handle setPersonPropertiesForFlags properly when personless
- 🛑 Posthog-js: handle manually calling $set properly when personless
- 🛑 Write RFC for client changes for personless events based on posthog-js, to be approved by mobile sdk teams
- 🆕 Get the sessions table v2 ingesting events

## OKR

1. Make querying fast enough for large customers
2. Do personless events work where necessary
3. Iterate on customer feedback
4. Product management work

### High priority
- Get session table v2 PR over the line
- Start backfilling, prioritising EU, and team 2 on US for dogfooding

### Stretch goal
- Get a versions of WA up that is terrifyingly fast because it can just use the sessions table + it can sample

**Comment by Twixes** [comment]
# Team Product Analytics

**Support hero:** @aspicer @skoob13

Time off: @Twixes the second week

## Retro

### High priority

- Getting rid of remaining legacy `filters` use (owner: @thmsobrmlr) – a bit left
- Query performance measurement (owner: @aspicer) – deprioritized overall, scope turned out to be much larger
- Project Environments finally (owner: @Twixes) – mostly deprioritized this week in favor of wrapping up persons-on-events/live events

### Low priority / side quests

- Insight background reloads monitoring/cleanup (@webjunkie) – pretty good state
- Onboarding Georgii (owner: @Twixes) - multiple breakdowns are progressing

## Hang over items from previous sprint

- Getting rid of remaining legacy `filters` use, the last bits (owner: @thmsobrmlr)
- Project Environments (owner: @Twixes)

## OKR

- HogQL-based querying
  - Convert the remaining legacy queries to HogQL and release to public (Thomas, Julian, ~~Marius~~)
    - 🟢 Insights – rolled out, although fixes needed to be rock-solid
    - ⚪ Cohorts
  - Remove legacy querying backend (Thomas, Julian)
    - 🟢 Clean up or rewrite dashboardLogic
    - 🟠 Convert filters to query (insights, notebooks, activity log, experiments)  👈 Thomas is on it
  - Missing Product Analytics features (Thomas, Julian)
    - 🟠 Breakdowns (multiple) in literally everything  👈  Georgiy is on it
    - 🟠 [Make a list](https://github.com/PostHog/meta/issues/200) based on GitHub issues from customer requests…
    - ⚪ Fix those issues
  - Missing HogQL features (~~Tom~~, ~~Marius~~)
    - 🟠 Type system, JSON 👈 Tom is on this, being in Data Warehouse
    - ⚪ Missing things when building funnels

- Querying and processing performance (Thomas, Julian) 
  - Global performance overview dashboards  👈 Sandy is on it
    - 🟠 Insights
    - ⚪ Exports
    - ⚪ Cohort recalculations
  - Query request tracing
    - ⚪ Possibly query runner Python optimizations
    - ⚪ Exports improvements
  - ⚪ Identify top 5 query optimizations in terms of impact

- Artificial Hog / Post Intelligence (Michael)
  - ⚪ Ask a question to get a magical insight (aware of your taxonomy)
  - ⚪ Figure out infra for upgrading queries and models
  - ⚪ Product-wide framework for opting into sharing with OpenAI

- Activation (side quest: Michael)
  - ⚪ Michael to work with Growth to identify optimizations to getting started with Product Analytics 👈 pending the product manager hire a bit

### High priority

- Experiments migrated from legacy trends/funnels to HogQL-based (owner: @thmsobrmlr)
- Getting rid of remaining legacy `filters` use continued (owner: @thmsobrmlr)
- Multiple breakdowns in Trends released to users (owner: @skoob13)
- Project Environments (owner: @Twixes)

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- [x] Q3 planning
- @raquelmsmith 
	- [x] Support for first week 
    - [x] Pricing page experiments - iterate here with cory and eli until it's done
    - [x] Stay on top of revenue issues
    - [x] Start working on toolbar dashboard template thing 
    - [x] Keep on top of personless comms and customer issues and metrics
    - [x] Lots of interviews...
- @zlwaterfield 
	- [ ] Complete _subscribe to all products_
		- [x]  frontend changes
		- [x] release under feature flag to new users
		- [ ] backfill existing users and communicate with them
		- [x] (if time permits - probably next sprint) cleanup! remove/clean single product subscribe code where we can.
	- [x] Start on the Stripe metadata changes - close RFC, updates to Zapier, work on backfill, etc.
	- [ ] re-run the plans map and compare with the new auto-cancel functionality

</details>


## Q2 Goals
✅=finished 🟡=in progress 🔴=won't finish


## This sprint

### High priority

- @raquelmsmith (support first week, on-call second week)
    - Personless events launch
	    - [ ] Oversee pricing calc changes, keep iterating until sales feels like it's good and we feel like it works for us as well
	    - [ ] If above is completed, make sure comms are sent out
	    - [ ] Make use-cases more apparent in onboarding
    - Data warehouse pricing
	    - [ ] Launch it for non-beta-users
    - Dashboard templates in onboarding
	    - [ ] Fix error that happens after creation, merge https://github.com/PostHog/posthog/pull/23069
	    - [ ] Add flow to launch toolbar from in-app
    - Hiring
    - Project-access-on-invites
    	- [ ] Do some digging to see what this entails (I don't think it will be involved or difficult)
- @zlwaterfield (on call first first week - support second week)
	- subscribe to all products
		- [ ] 	Run backfill for subscribe to all products and notify users
		- [ ] 	Remove feature flag code for subscribe to all products and cleanup code
	- stripe startup metadata
		- [ ] 	Finish stripe metadata clean - 20-30 left to manual fixes + a few hours of manual checks
		- [ ] 	Build a startup plan dashboard in dashboard in PostHog
	- misc
		- [ ] 	Think through "Free / paid - same feature-set"
		- [ ] 	Add at least one E2E SAML test
		- [ ] 	Run backfill for starting/ending backfill bug
		- [ ] 	Re-run the plans map compare
		- [ ] Deprecate billing v2 (PR done just need to merge)
